crates/sui-core/src/state_accumulator.rs

```diff
@@ -43,18 +44,47 @@ impl StateAccumulator {
 
         let mut acc = Accumulator::default();
 
+        // process insertions to the set
         acc.insert_all(
             effects
                 .iter()
                 .flat_map(|fx| {
                     fx.created
                         .clone()
                         .into_iter()
-                        .map(|(obj_ref, _)| obj_ref.2)
+                        .map(|(oref, _)| oref.2)
+                        .chain(fx.wrapped.clone().into_iter().map(|oref| oref.2))
+                        .chain(fx.unwrapped.clone().into_iter().map(|(oref, _)| oref.2))
+                        .chain(fx.mutated.clone().into_iter().map(|(oref, _)| oref.2))
                         .collect::<Vec<ObjectDigest>>()
                 })
                 .collect::<Vec<ObjectDigest>>(),
         );
+
+        // get all modified_at_versions for the fx
+        let modified_at_version_keys = effects
+            .iter()
+            .flat_map(|fx| {
+                fx.modified_at_versions
+                    .clone()
+                    .into_iter()
+                    .map(|(id, seq_num)| ObjectKey(id, seq_num))
+                    .collect::<Vec<ObjectKey>>()
+            })
+            .collect::<Vec<ObjectKey>>();
+
+        let modified_at_digests = self
+            .authority_store
+            .perpetual_tables
+            .objects
+            .multi_get(modified_at_version_keys)?
+            .iter()
+            // TODO(william) try to avoid hashing every object here by adding digest
```

Comment:
this is only necessary for dynamic loads, right? otherwise you can get the digest from the tx input objects, right?

Reply:
Not sure. Where are tx input objects made available within txn effects?